extensions/ql-vscode/src/pure/errors.ts

```diff
@@ -54,19 +54,35 @@ export function redactableError(
   ...values: unknown[]
 ): RedactableError;
 export function redactableError(
-  error: Error,
+  error: ErrorLike,
 ): (strings: TemplateStringsArray, ...values: unknown[]) => RedactableError;
 
 export function redactableError(
-  errorOrStrings: Error | TemplateStringsArray,
+  errorOrStrings: ErrorLike | TemplateStringsArray,
   ...values: unknown[]
 ):
   | ((strings: TemplateStringsArray, ...values: unknown[]) => RedactableError)
   | RedactableError {
-  if (errorOrStrings instanceof Error) {
+  if (isErrorLike(errorOrStrings)) {
     return (strings: TemplateStringsArray, ...values: unknown[]) =>
       new RedactableError(errorOrStrings, strings, values);
   } else {
     return new RedactableError(undefined, errorOrStrings, values);
   }
 }
+
+export interface ErrorLike {
+  message: string;
+  stack?: string;
+}
+
+function isErrorLike(error: any): error is ErrorLike {
+  if (
+    error.message !== undefined &&
+    typeof error.message === "string" &&
```

Comment:
Minor:

```suggestion
    typeof error?.message === "string" &&
```

Reply:
Thinking about this more, I think we can simplify this to just `typeof error.message === "string"` and the `?` is unnecessary. That question mark is purely a typescript thing and is removed when transpiling to javascript, but the type of `error` here is `any` so there's no typescript type checking anyway. So we can just remove it. You can safely do `typeof undefined` and it'll return `"undefined"`.